### ISSUE TITLE
Resource tree with duplicate container or missing page

### ISSUE THREAD

**Issue by funkbilly** [modxcms/fred#67]
Hi, thanks for Fred, it's such a great addition to ModX!

I have been in trouble though (on the Fred frontend side menu, site tab), because one of the root resource container (french version of the site) was duplicated and with some children resources missing in it. The second root resource container (the english version of the site) was fine with all its resources. 
**It was looking something like this:**

- French version (id = 2)
  - | Page 6
  - | Page 7
  - | Page 8
- French version (id = 2)
  - | Page 6
  - | Page 7
  - | Page 8
- English version (id = 3)
  - | Page 1
  - | Page 2
  - | Page 3
  - | Page 4
  - | Page 5
  - | Page 6
  - | Page 7
  - | Page 8

**Instead of:**

- French version (id = 2)
  - | Page 1
  - | Page 2
  - | Page 3
  - | Page 4
  - | Page 5
  - | Page 6
  - | Page 7
  - | Page 8
- English version (id = 3)
  - | Page 1
  - | Page 2
  - | Page 3
  - | Page 4
  - | Page 5
  - | Page 6
  - | Page 7
  - | Page 8

**Until I add one line in**
core / components / fred / model / fred / src / Endpoint / Ajax / GetResourceTree.php

```php
protected function handleResource($resource, $isFred) {
    $pageFormatted = [
         'id' => $resource->id,
         'pagetitle' => $resource->pagetitle,
         'children' => [],
...
```

so it goes like

```php
protected function handleResource($resource, $isFred) {
    if(isset($this->map[$resource->id])) return;
    $pageFormatted = [
         'id' => $resource->id,
         'pagetitle' => $resource->pagetitle,
         'children' => [],
...
```

I wanted to share this. Does it make sense?

Cheers

**Comment by theboxer** [comment]
Hello,
what version of Fred are you using?

**Comment by theboxer** [comment]
@funkbilly what version of Fred are you using?

**Comment by funkbilly** [comment]
Hi, sorry for my late reply. I'm using Fred 1.0.0-beta12

**Comment by theboxer** [comment]
Could you provide more details about all the resources? Are they sharing Template? Are they all fred resources?

And the actions you are doing? You're trying to duplicate the `French version (id = 2)` from frontend?

**Comment by funkbilly** [comment]
The resources are in the "web" context. Exactly same parameters, same template, and yes, editable with Fred. I did not even try to duplicate or create or delete resource using Fred. The resources were existing before I install Fred and I did not change anything with the tree structure.

I went into the code when I was sure that the resources were exactly the same. Honestly, I did not try to investigate, but I felt that the children were not populated correctly in the map, or like the same resource was defined twice, but with less data at the 2nd pass. I added this line in order to prevent this, and it did the trick.
